mempool/v1/mempool.go

```diff
@@ -791,16 +791,12 @@ func (txmp *TxMempool) purgeExpiredTxs(blockHeight int64) {
 		next := cur.Next()
 
 		w := cur.Value.(*WrappedTx)
-		if txmp.config.TTLNumBlocks > 0 && (blockHeight-w.height) > txmp.config.TTLNumBlocks {
+		if txmp.config.TTLNumBlocks > 0 && (blockHeight-w.height) > txmp.config.TTLNumBlocks ||
+			txmp.config.TTLDuration > 0 && now.Sub(w.timestamp) > txmp.config.TTLDuration {
 			txmp.removeTxByElement(cur)
 			txmp.cache.Remove(w.tx)
-			txmp.metrics.EvictedTxs.With(mempool.TypeLabel, mempool.EvictedTxExpiredBlocks).Add(1)
 			txmp.evictedTxs.Push(w.tx)
-		} else if txmp.config.TTLDuration > 0 && now.Sub(w.timestamp) > txmp.config.TTLDuration {
-			txmp.removeTxByElement(cur)
-			txmp.cache.Remove(w.tx)
-			txmp.evictedTxs.Push(w.tx)
-			txmp.metrics.EvictedTxs.With(mempool.TypeLabel, mempool.EvictedTxExpiredTime).Add(1)
+			txmp.metrics.ExpiredTxs.Add(1)
```

Comment:
[nit][no change needed] seems a bit confusing to have evicted / expired metrics but only have one field in mempool called `evictedTxs` that both evicted + expired transactions get added to.

Reply:
I can see that. For the metrics we care about whether a transaction was evicted because of being too low priority or expired because of a TTL. For the user that uses the `TxStatus` endpoint that uses the `evictedTxs` map, we assume they don't care about this distinction